src/ar/RewayatClub.lua

```diff
@@ -0,0 +1,128 @@
+    arrFlatten(arr)
+    return results
+end
+
+return {
+    id = 3302,
+    name = "RewayatClub - نادي الروايات",
+    baseURL = baseURL,
+    hasSearch = true,
+    listings = {
+        Listing("Novel List",true,function (data)
+			local res = Request(GET(baseURL .. "/api/novels/?page=" .. data[PAGE]+1)):body():string()
+
+            local d = json.decode(res)
+
+            return map(d.results,function(v) 
+                    return Novel {
+                        title = v.english,
+                        imageURL = baseUrlApi .. "/".. v.poster_url,
+                        link = baseURL .. "/novel/" .. v.slug
+                    }
+                end)
+            end)
+            ,Listing("Latest",true,function (data)
+			    local res = Request(GET(baseURL .. "/api/chapters/weekly/list/?page=" .. data[PAGE]+1)):body():string()
+
+                local d = json.decode(res)
+    
+                return map(d.results ,function (v)
+                        return Novel {
+                            title = v.novel.english,
+                            imageURL = baseUrlApi .. "/" .. v.novel.poster_url,
+                            link = baseURL .. "/novel/" .. v.novel.slug
+                        }    
+                    end)
+                end)
+    },
+    parseNovel = function (novelURL)
+        local url1 = novelURL:gsub("/novel/","/api/novels/")
+        local res = Request(GET(url1)):body():string()
+
+        local d = json.decode(res)
+
+        
+        local novelInfo = NovelInfo()
+        novelInfo:setTitle(d.english)
+        novelInfo:setImageURL(baseUrlApi .. d.poster_url)
+        novelInfo:setDescription(d.english)
+
+        local status = d.get_novel_status
+        novelInfo:setStatus(NovelStatus(status == "مكتملة" and 1 or status == "متوقفة" and 2 or status == "مستمرة" and 0 or 3))
+        novelInfo:setGenres(map(d.genre,function (v)
+            return v.english
+        end))
+
+        local url2 = novelURL:gsub("/novel/","/api/chapters/") .. "/all"
+		local res2 = Request(GET(url2)):body():string()
+
+        local chaptersJson = json.decode(res2)
+
+        local chapterList = AsList(map(chaptersJson,function (v)
+            local c = NovelChapter()
+            c:setLink(novelURL.."/" .. v.number)
+            c:setTitle(v.title)
+            c:setOrder(v.number)
+            return c
+        end))
+        novelInfo:setChapters(chapterList)
+        
+        return novelInfo
+        
+    end,
+    getPassage = function (chapterURL)
+        local url = chapterURL:gsub("/novel/","/api/chapters/")
+		local res = Request(GET(url)):body():string()
+        local resJson = json.decode(res)
+
+        return table.concat(map(resJson.content,function (v) 
+                return table.concat(map(v,function (c) 
+                    -- local temp = c:gsub("<br/>","<b/r/>")
+                        return Document(c):text()
+                    end),"\n")
+            end),"\n"):gsub("\n\n","\n")
```

Comment:
Since the `Document(c)` here implies the API returns HTML, I suggest you use `ChapterType.HTML` instead, which would simplify this by a lot.

Reply:
I'm not getting the ChapterType.HTML part

Reply:
It's fine. Someone else can do it if needed.

Reply:
That's a nonreversible change for users. I suggest against.